app/scripts/modules/core/src/pipeline/executions/Executions.tsx

```diff
@@ -185,43 +186,48 @@ export class Executions extends React.Component<IExecutionsProps, IExecutionsSta
     );
   }
 
-  public handleTransitionSuccess(transition: Transition): void {
+  // if we're navigating to a different execution on the same page, scroll the new execution into view
+  // or, if we are navigating back to the same execution after scrolling down the page, scroll it into view
+  // but don't scroll it into view if we're navigating to a different stage in the same execution
+  private maybeScrollToExecution(transition: Transition): void {
     const toParams = transition.params('to');
     const fromParams = transition.params('from');
-    // if we're navigating to a different execution on the same page, scroll the new execution into view
-    // or, if we are navigating back to the same execution after scrolling down the page, scroll it into view
-    // but don't scroll it into view if we're navigating to a different stage in the same execution
-    let shouldScroll = false;
-    if (
-      transition.to.name.indexOf(transition.from.name) === 0 &&
+    const shouldScroll =
+      toParams.executionId &&
       toParams.application === fromParams.application &&
-      toParams.executionId
-    ) {
-      shouldScroll = true;
-      if (toParams.executionId === fromParams.executionId && toParams.details) {
-        if (
-          toParams.stage !== fromParams.stage ||
-          toParams.step !== fromParams.step ||
-          toParams.details !== fromParams.details
-        ) {
-          shouldScroll = false;
-        }
-      }
-    }
+      toParams.executionId !== fromParams.executionId;
+
     if (shouldScroll) {
       this.scrollIntoView(0);
     }
   }
 
+  // If the execution filter parameters changed, refresh executions
+  private maybeRefreshExecutions(transition: Transition): void {
+    const toParams = transition.params('to');
+    const fromParams = transition.params('from');
+    const compareParams = ['q', 'pipeline', 'status'];
```

Comment:
While `pipeline` and `status` should force us to call back to the server, `q` filtering is done in-page - we don't need to fetch new data from the server to handle it.

Reply:
ah, thanks 👍